pymdp/learning.py

```diff
@@ -73,34 +73,45 @@ def update_state_transition_dirichlet_f(pB_f, actions_f, joint_qs_f, lr=1.0):
 def update_state_transition_dirichlet(pB, B, joint_beliefs, actions, *, num_controls, lr, factors_to_update='all'):
     """"
     Update posterior Diriichlet parameters of the state transition likelihood model (B) given the joint beliefs over hidden states and actions.
+
+    Supports selective learning of only particular hidden state factors via the `factors_to_update` argument, which can either be "all" or a list of factor indices to update.
     """
     nf = len(pB)
 
     actions_onehot_fn = lambda f, dim: nn.one_hot(actions[..., f], dim, axis=-1)
 
-    update_B_f_fn = lambda pB_f, joint_qs_f, f, na: None if pB_f is None else update_state_transition_dirichlet_f(
-        pB_f, actions_onehot_fn(f, na), joint_qs_f, lr=lr
-    )
+    def update_B_f_fn(pB_f, joint_qs_f, f, na):
+       """ 
+       Conditionally-update the Dirichlet posterior over a given single factor's B parameters
+       Updating is conditional upon the value of `f`: if the factor index (f) is greater than -1, then use the value of f as the factor index
+       to create the appropriate one-hot representation of the action corresponding to the the f-th control factor and perform the update. Otherwise, do not perform t he update
```

Comment:
```suggestion
       to create the appropriate one-hot representation of the action corresponding to the the f-th control factor and perform the update. Otherwise, do not perform the update
```